### PR TITLE
Preserve tensor learning rates across scheduler updates

### PR DESCRIPTION
**Pull request by n33levo**
## What it is

DeepSpeed's tensor learning-rate support currently replaces an optimizer's LR tensor on every scheduler update. A scalar `float64` LR tensor becomes a new one-dimensional tensor during scheduler initialization, changing its identity, shape, and initially its dtype. This also leaves any caller-held reference to the supplied LR tensor pointing at the stale value.

The root cause is `update_lr()` constructing `tensor([lr])` instead of updating the tensor supplied through the optimizer. This is a follow-up to the tensor LR support added in #7633.

## How it works

- Fill the optimizer's existing LR tensor in place, matching PyTorch scheduler behavior.
- Squeeze a calculated one-element tensor to the scalar value expected by `fill_`, covering both zero-dimensional and one-element LR tensors.
- Snapshot tensor base LRs in `WarmupCosineLR` before initialization updates the optimizer tensor, preserving the original value used by later schedule steps.
- Leave the scalar LR assignment path unchanged.
- Add `WarmupLR` and `WarmupCosineLR` regression tests that check object identity, shape, dtype, initialization, and scheduled values.

## E2E Top-hatting

On current `master`, a zero-dimensional `float64` LR tensor is replaced by a different one-dimensional tensor. With this change, both zero-dimensional and one-element LR tensors retain their original identity, shape, and dtype while reaching the expected warmup values. `WarmupCosineLR` also retains an independent base value, preventing its schedule from remaining at zero after initialization.

A clean GitHub-hosted Ubuntu 24.04 CPU run used Python 3.11 and PyTorch 2.10:

- `pytest --forked -n 4 unit/runtime/test_lr_schedulers.py --torch_ver=2.10`
- **70 passed** ([run](https://github.com/n33levo/DeepSpeed/actions/runs/30714252928))

## Checks

- `pre-commit run --files deepspeed/runtime/lr_schedules.py tests/unit/runtime/test_lr_schedulers.py`
- All formatting, lint, license, spelling, and custom Torch checks passed.

### CODE REVIEW

File: deepspeed/runtime/lr_schedules.py [hunk 1]
```diff
@@ -250,10 +250,11 @@ def get_lr_from_config(config):
 
 def update_lr(param_groups, lrs):
     for param_group, lr in zip(param_groups, lrs):
-        # new LR should match the type of current LR for scalar and Tensor LR support
         if is_tensor(param_group['lr']):
-            lr = tensor([lr], device=param_group['lr'].device)
-        param_group['lr'] = lr
+            lr = lr.squeeze() if is_tensor(lr) else lr
+            param_group['lr'].fill_(lr)
```
[the remark this answers is not in the document]
Comment (n33levo) [reply deepspeed/runtime/lr_schedules.py:255]: Good catch. I now clone tensor-valued base LRs before initialization and added regression coverage for both 0-D and one-element LR tensors. The updated Ubuntu scheduler suite passes all 70 tests.

File: deepspeed/runtime/lr_schedules.py [hunk 2]
```diff
@@ -13,7 +13,7 @@
 from torch.optim import Optimizer
 import math
 from deepspeed.utils import logger
-from torch import tensor, is_tensor
+from torch import is_tensor
```
[the remark this answers is not in the document]
Comment (n33levo) [reply deepspeed/runtime/lr_schedules.py:16]: The original commit already has the required sign-off, and the follow-up commit is signed off as well. DCO is passing.

**Review by sfc-gh-truwase — APPROVED** [review]
